### PR TITLE
documentation: user deletion API

### PR DESCRIPTION
**Pull request by fflorent**
I see that there is an endpoint for deleting users, but it is undocumented.

Still this endpoint can be useful, and I am interested in it (also in evolving it). I am documenting for now its current behavior, if you think it is ready to be officially proposed.

### CODE REVIEW

File: api/grist.yml [hunk 1]
```diff
@@ -1124,6 +1157,10 @@ tags:
     description: "Documents may include attached files. Data records can refer to these using a column of type `Attachments`."
   - name: webhooks
     description: "Document changes can trigger requests to URLs called webhooks."
+  - name: sql
+    description: "Sql endpoint to query data from documents."
+  - name: users
+    description: "Users of Grist"
```
Comment (vviers) [inline api/grist.yml:1163]:
```suggestion
    description: "Grist users."
```

File: api/grist.yml [hunk 2]
```diff
@@ -1104,6 +1104,39 @@ paths:
             application/json:
               schema:
                 $ref: "#/components/schemas/SqlResultSet"
+  /users/{userId}:
+    delete:
+      tags:
+        - users
+      summary: "Delete a user from Grist"
+      description: |
+        Also delete the user's personal org (aka team site).
```
Comment (vviers) [inline api/grist.yml:1113]:
```suggestion
        This action also deletes the user's personal organisation and all the workspaces and documents it contains.
```

File: api/grist.yml [hunk 3]
```diff
@@ -1104,6 +1104,39 @@ paths:
...
+        Currently only the user themselves is allowed to delete their own account.
```
Comment (vviers) [inline api/grist.yml:1114]:
```suggestion
        Currently, only the users themselves are allowed to delete their own accounts.
```

File: api/grist.yml [hunk 4]
```diff
@@ -1104,6 +1104,39 @@ paths:
...
+
+        ⚠️ **This cannot be undone, please be cautious when using this endpoint** ⚠️
```
Comment (vviers) [inline api/grist.yml:1116]:
```suggestion
        ⚠️ **This action cannot be undone, please be cautious when using this endpoint** ⚠️
```

File: api/grist.yml [hunk 5]
```diff
@@ -1104,6 +1104,39 @@ paths:
...
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - name
+              properties:
+                name:
+                  type: string
+                  description: "The user's name to delete (for confirmation, to avoid deleting the wrong account)."
```
Comment (vviers) [inline api/grist.yml:1129]: Not an issue, but intuitively I would think that passing an email address would be easier than passing a user name 🤷🏼

**Review by vviers — APPROVED** [review]
LGTM, minor edits suggested

File: api/grist.yml (hunk 5 above)
Comment (fflorent) [reply api/grist.yml:1129]: Good point, especially the normalized one. Though if we take a look at the [database structure](https://github.com/gristlabs/grist-core/blob/main/documentation/database.md#the-schema), the email is part of the *logins* table. Though there should only be a user per email address, so I feel like this should be OK.

Reported here: https://github.com/gristlabs/grist-core/issues/1134

**Review by paulfitz — APPROVED** [review]
Thanks you @fflorent !

**Pull request merged**